social_core/backends/vk.py

```diff
@@ -12,6 +12,46 @@
 from .oauth import BaseOAuth2
 from ..exceptions import AuthTokenRevoked, AuthException
 
+DEFAULT_API_VERSION = '5.131'
+
+
+class VKApiCaller:
+    API_BASE_URL = 'https://api.vk.com'
+
+    def get_api_url(self, method, data):
+        if 'access_token' in data:
+            return self.API_BASE_URL + '/method/' + method
+        return self.API_BASE_URL + '/api.php'
+            
+    def make_sure_access_token_exists(self, method, data):
+        if 'access_token' in data:
+            return data
```

Comment:
Seems like access_token is always present, maybe the code can be simplified?

Reply:
I'm afraid it's not, when we call the method from VKAppOAuth2, we don't have access_token.

Reply:
Ah, I missed the `isAppUser` call when reviewing, and lack of test coverage there made me think it is not used. Can you add tests for that?

Reply:
Sure, I'll add.

Reply:
@blacktrub Any chance to get to this? :smile: